app/cdash/include/autoremove.php

```diff
@@ -79,25 +83,21 @@ function removeFirstBuilds(int $projectid, int $days, int $maxbuilds, bool $forc
     $currentdate = time() - 3600 * 24 * $days;
     $startdate = date(FMT_DATETIME, $currentdate);
 
-    add_log('about to query for builds to remove', 'removeFirstBuilds');
-    $builds = DB::select('
-                   SELECT id
-                   FROM build
-                   WHERE
-                       parentid IN (0, -1)
-                       AND starttime < ?
-                       AND projectid = ?
-                   ORDER BY starttime ASC
-                   LIMIT ?
-               ', [$startdate, intval($projectid), $maxbuilds]);
+    Log::info('about to query for builds to remove');
 
+    $builds = Build::whereIn('parentid', [0, -1])
+        ->where('starttime', '<', $startdate)
+        ->where('projectid', '=', $projectid)
+        ->orderBy('starttime')
+        ->limit($maxbuilds)
+        ->get();
     $buildids = [];
     foreach ($builds as $build) {
         $buildids[] = (int) $build->id;
     }
```

Comment:
You can use `->pluck('id')->toArray()` to get an array of integer IDs instead of querying entire models and then looking over the results.